src/os/bluestore/BlueStore.h

```diff
@@ -258,12 +258,35 @@ class BlueStore : public ObjectStore,
       assert(writing.empty());
     }
 
-    void _add_buffer(Cache* cache, Buffer *b, int level, Buffer *near) {
+    void _add_buffer(Cache* cache,
+                     Buffer *b,
+                     int level,
+                     Buffer *near,
+                     bool maybe_require_sort = false) {
       cache->_audit("_add_buffer start");
       buffer_map[b->offset].reset(b);
       if (b->is_writing()) {
 	b->data.reassign_to_mempool(mempool::mempool_bluestore_writing);
-        writing.push_back(*b);
+        if (writing.empty() || !maybe_require_sort ||
```

Comment:
why do we need that maybe_require_sort flag, wouldn't the comparison against the last item's seq enough to determine the need for sorted insert?